### PR TITLE
Fix `ConvertType`

### PR DESCRIPTION
**Pull request by LeviPesin**
Related issue: https://github.com/mrdoob/three.js/pull/24964#issuecomment-1315622236

**Description**

I think this should fix it...

### CODE REVIEW

File: examples/jsm/nodes/utils/JoinNode.js [hunk 1]
```diff
@@ -2,17 +2,18 @@ import TempNode from '../core/Node.js';
...
 
 		this.nodes = nodes;
 
 	}
 
 	getNodeType( builder ) {
 
-		return builder.getTypeFromLength( this.nodes.reduce( ( count, cur ) => count + builder.getTypeLength( cur.getNodeType( builder ) ), 0 ) );
+		const defaultType = builder.getTypeFromLength( this.nodes.reduce( ( count, cur ) => count + builder.getTypeLength( cur.getNodeType( builder ) ), 0 ) );
+		return this.nodeType || defaultType;
```
Comment (LeviPesin) [inline examples/jsm/nodes/utils/JoinNode.js:16]: Alternatively, we can change the type based on the componentType of the type of the first node?

**Comment by LeviPesin** [comment]
It seems that actually that bug is two bugs, and this PR fixes only one of them... With this PR WebGPU complains on `uvec4( index, index, index, uint( 1 ) )` like so:

```
1 error(s) generated while compiling the shader:
:32:49 error: no matching initializer for vec4<u32>(vec3<u32>, abstract-float)

10 candidate initializers:
  vec4(xyz: vec3<T>, w: T) -> vec4<T>  where: T is abstract-int, abstract-float, f32, f16, i32, u32 or bool
  vec4(xy: vec2<T>, zw: vec2<T>) -> vec4<T>  where: T is abstract-int, abstract-float, f32, f16, i32, u32 or bool
  vec4(x: T, zyw: vec3<T>) -> vec4<T>  where: T is abstract-int, abstract-float, f32, f16, i32, u32 or bool
  vec4(vec4<T>) -> vec4<T>  where: T is abstract-int, abstract-float, f32, f16, i32, u32 or bool
  vec4(T) -> vec4<T>  where: T is abstract-int, abstract-float, f32, f16, i32, u32 or bool
  vec4<T>() -> vec4<T>  where: T is f32, f16, i32, u32 or bool
  vec4(xy: vec2<T>, z: T, w: T) -> vec4<T>  where: T is abstract-int, abstract-float, f32, f16, i32, u32 or bool
  vec4(x: T, yz: vec2<T>, w: T) -> vec4<T>  where: T is abstract-int, abstract-float, f32, f16, i32, u32 or bool
  vec4(x: T, y: T, zw: vec2<T>) -> vec4<T>  where: T is abstract-int, abstract-float, f32, f16, i32, u32 or bool
  vec4(x: T, y: T, z: T, w: T) -> vec4<T>  where: T is abstract-int, abstract-float, f32, f16, i32, u32 or bool

5 candidate conversions:
  vec4<T>(vec4<U>) -> vec4<f32>  where: T is f32, U is abstract-int, abstract-float, i32, f16, u32 or bool
  vec4<T>(vec4<U>) -> vec4<f16>  where: T is f16, U is abstract-int, abstract-float, f32, i32, u32 or bool
  vec4<T>(vec4<U>) -> vec4<i32>  where: T is i32, U is abstract-int, abstract-float, f32, f16, u32 or bool
  vec4<T>(vec4<U>) -> vec4<u32>  where: T is u32, U is abstract-int, abstract-float, f32, f16, i32 or bool
  vec4<T>(vec4<U>) -> vec4<bool>  where: T is bool, U is abstract-int, abstract-float, f32, f16, i32 or u32

  NodeBuffer_291.nodeUniform0[ instanceIndex ] = vec4<u32>( vec4<u32>( instanceIndex, instanceIndex, instanceIndex, 1u ).xyz, 1.0 );
                                                 ^^^^
```
Note there are no `vec4<f32>` there anymore.

File: examples/jsm/nodes/utils/JoinNode.js (hunk 1 above)
Comment (sunag) [reply examples/jsm/nodes/utils/JoinNode.js]: Yes, I just think it doesn't need to compute the `defaultType` if `this.nodeType` is not null.

**Review by sunag — APPROVED** [review]

**Comment by sunag** [comment]
Thanks! :)

**Pull request merged**